jina/flow/__init__.py

```diff
@@ -251,6 +251,19 @@ def needs(self, needs: Union[Tuple[str], List[str]],
             raise FlowTopologyError('no need to wait for a single service, need len(needs) > 1')
         return self.add(name=name, needs=needs, pod_role=PodRoleType.JOIN, *args, **kwargs)
 
+    def needs_all(self, copy_flow: bool = True,
+                  name: str = 'joiner', *args, **kwargs) -> 'Flow':
+        """
+        Collect all hanging Pod so far and add a blocker to the flow, wait until all handing peas completed.
+        :param copy_flow: when set to true, then always copy the current flow and do the modification on top of it then return, otherwise, do in-line modification
+        :param name: the name of this joiner, by default is ``joiner``
+        :return: the modified flow
+        """
+        op_flow = copy.deepcopy(self) if copy_flow else self
+        needs = _hanging_pods(op_flow)
+
+        return self.needs(needs=needs, name=name, *args, **kwargs)
```

Comment:
This won't work, if there is only one hanging pod. Thus, you can either add a flag `accept_one_pod` to the `needs` function in order to reuse `needs` or directly call `add` here.

Reply:
I have updated the code to fix these issues accordingly and also added required unit tests, please let me know if that works. Thanks!

Reply:
Looks great so far. One tiny cleanup: Could you switch the order of variables in the calling of `add` and `needs` to be the same? That'll improve readability.

Reply:
Sure! Thank you for the advice! 
I have made the change to rearrange them and keep it consistent. 
I have also thought a little about it before and the reason I arranged the order of variables this way is that I wanted to keep it consistent with current conventions. (e.g. In the unit test test_flow_arbitrary_needs(), .add() is called as (name, needs) and .needs is called as (needs, name).